calcom/cal.com
# fix: GET /v2/schedules incorrectly returning schedules from other accounts #17804

## What does this PR do?
fix #17804 

The `/v2/schedules` endpoint was incorrectly returning schedules for users outside the authenticated user's organization. This PR fixes the issue by:

- Ensuring only schedules from users within the same organization are returned
- Implementing checks for both direct organization membership and profiles
- Excluding schedules from locked or deleted users
- Maintaining proper authentication context throughout the request

## Technical Implementation
Modified the `getSchedulesByUserId` method in the `SchedulesRepository_2024_06_11` class to:
- Add organization membership validation in the schedule retrieval logic
- Implement profile-based organization membership checks
- Add filters to exclude schedules from locked/inactive users
- Ensure proper data isolation between organizations

## Testing Steps
1. Log in as a user in an organization
2. Access the `/v2/schedules` endpoint
3. Verify that only schedules from users in the same organization are returned
4. Verify that schedules from locked/deleted users are not returned

## Issue References
- Fixes #17804 (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description) (issue number not given)

## Mandatory Tasks (DO NOT REMOVE)

- [✔️] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [✔️] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [✔️] I confirm automated tests are in place that prove my fix is effective or that my feature works.
